api/src/main/java/org/openmrs/module/emrapi/adt/AdtServiceImpl.java

```diff
@@ -1061,14 +1061,33 @@ else if (encounter.getEncounterType().equals(dischargeEncounterType)) {
             }
         }
 
-        List<InpatientAdmission> ret = new ArrayList<>();
-        for (InpatientAdmission admission : m.values()) {
-            if (criteria.getCurrentInpatientLocations() == null || criteria.getCurrentInpatientLocations().contains(admission.getCurrentInpatientLocation())) {
-                if (criteria.isIncludeDischarged() || !admission.isDischarged()) {
-                    ret.add(admission);
-                }
+        // Filter out any admissions that do not match the search criteria
+        List<Integer> visitIds = new ArrayList<>();
+        for (Iterator<Visit> i = m.keySet().iterator(); i.hasNext(); ) {
+            Visit visit = i.next();
+            InpatientAdmission admission = m.get(visit);
```

Comment:
I think it probably makes more sense to write this in terms of an `EntrySet` since we only care about values:

```suggestion
        for (Iterator<EntrySet<Visit, InpatientAdmission>> i = m.entrySet().iterator(); i.hasNext(); ) {
            EntrySet<Visit, InpatientAdmission> entry = i.next();
            InpatientAdmission admission = entry.getValue();
```

Reply:
We could even do:

```suggestion
        for (Iterator<EntrySet<Visit, InpatientAdmission>> i = m.keySet().iterator(); i.hasNext(); ) {
            InpatientAdmission admission =  i.next().getValue();
```

Reply:
Fixed, as suggested.